File: doc/codes/python/examples/api/compute/modulation/square.py
from autd3.units import Hz
from autd3.value import SamplingConfig
from autd3_modulation import SquareOption, modulation_buffer, square

freq = 150.0 * Hz
low = 0x00
high = 0xFF
duty = 0.5
sampling_config = SamplingConfig.FREQ_4K
option = (
    # ANCHOR: option
    SquareOption(
        low,
        high,
        duty,
        sampling_config,
    )
    # ANCHOR_END: option
)
dst = modulation_buffer()
# ANCHOR: api
square(freq, option, dst)
# ANCHOR_END: api

File: doc/codes/csharp/examples/api/compute/modulation/square.cs
using AUTD3;
using static AUTD3.Units;

namespace DocSamples.ApiComputeModulationSquare;

internal static class Sample
{
    internal static void Run()
    {
        var freq = 150 * Hz;
        var low = byte.MinValue;
        var high = byte.MaxValue;
        var duty = 0.5f;
        var samplingConfig = SamplingConfig.Freq4k;
        var option =
            // ANCHOR: option
            new SquareOption(
                low,
                high,
                duty,
                samplingConfig
            )
            // ANCHOR_END: option
            ;
        var dst = Modulation.ModulationBuffer();
        // ANCHOR: api
        Modulation.Square(freq, option, dst);
        // ANCHOR_END: api
    }
}
